ui/src/store/features/bunny-cdn/async-thunk/verify-ap.ts

```diff
@@ -0,0 +1,34 @@
+import { createAsyncThunk } from '@reduxjs/toolkit';
+
+import { verifyBunnyCDNMock } from '@/mocks';
+import { RootState } from '@/store';
+import { AppLog } from '@/utils';
+
+import { bunnyCDNActions } from '../bunny-cdn-slice';
+
+export const verifyAP = createAsyncThunk<void, string>(
+  'BunnyCDN/VerifyAP',
+  async (domain, { dispatch, getState }) => {
+    const { state } = (getState() as RootState).bunnyCDN;
+
+    if (state === 'loading') return;
+
+    try {
+      dispatch(bunnyCDNActions.setState('loading'));
+
+      const verifyAPState = await verifyBunnyCDNMock(domain);
+
+      if (verifyAPState) dispatch(bunnyCDNActions.setState('success'));
+      else {
+        AppLog.errorToast('We couldn`t verify the domain. Please, try again');
+        dispatch(bunnyCDNActions.setState('failed'));
+      }

```

Comment:
Throwing here helps you to not twice the code:
```suggestion
      if (verifyAPState) return dispatch(bunnyCDNActions.setState('success'));
      throw new Error('Invalid AP state');
```